tests/rc_string_ut/rc_string_ut.c

```diff
@@ -218,6 +241,174 @@ TEST_FUNCTION(when_underlying_calls_fail_rc_string_create_also_fails)
+TEST_FUNCTION(when_vsnprintf_determine_written_characters_length_fail_rc_string_create_with_format_also_fails)
+{
+    // arrange
+    STRICT_EXPECTED_CALL(mocked_vsnprintf(IGNORED_ARG, IGNORED_ARG, IGNORED_ARG, IGNORED_ARG))
+        .SetReturn(-1);
+
+    // act
+    THANDLE(RC_STRING) rc_string = rc_string_create_with_format("hell%c", 'o');
+
+    // assert
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+    ASSERT_IS_NULL(rc_string);
+
+}
+
+/* Tests_SRS_RC_STRING_07_002: [ Otherwise `rc_string_create_with_format` shall determine the total number of characters written using `vsnprintf` with the variable number of arguments. ]*/
+/* Tests_SRS_RC_STRING_07_004: [ `rc_string_create_with_format` shall allocate memory for the `THANDLE(RC_STRING)`and the number of bytes for the resulting formatted string. ]*/
+/* Tests_SRS_RC_STRING_07_005: [ `rc_string_create_with_format` shall fill in the bytes of the string by using `vsnprintf`. ]*/
+/* Tests_SRS_RC_STRING_07_007: [ `rc_string_create_with_format` shall succeed and return a non - `NULL` handle. ]*/
+TEST_FUNCTION(rc_string_create_with_format_with_empty_string_succeeds)
+{
+    // arrange
+    STRICT_EXPECTED_CALL(mocked_vsnprintf(IGNORED_ARG, IGNORED_ARG, IGNORED_ARG, IGNORED_ARG));
+    STRICT_EXPECTED_CALL(malloc_flex(IGNORED_ARG, IGNORED_ARG, sizeof(char)));
+    STRICT_EXPECTED_CALL(mocked_vsnprintf(IGNORED_ARG, IGNORED_ARG, IGNORED_ARG, IGNORED_ARG));
+
+    // act
+    THANDLE(RC_STRING) rc_string = rc_string_create_with_format("");
+
+    // assert
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+    ASSERT_IS_NOT_NULL(rc_string);
+    ASSERT_ARE_EQUAL(char_ptr, "", rc_string->string);
+
+    // cleanup
+    THANDLE_ASSIGN(RC_STRING)(&rc_string, NULL);
+}
+
+/*Tests_SRS_RC_STRING_07_006: [ If `vsnprintf` failed to get the resulting formatted string, `rc_string_create_with_format` shall fail and return `NULL`. ]*/
+TEST_FUNCTION(when_vsnprintf_get_resulting_formatted_string_fail_rc_string_create_with_format_also_fails)
+{
+    //// arrange
+    STRICT_EXPECTED_CALL(mocked_vsnprintf(IGNORED_ARG, IGNORED_ARG, IGNORED_ARG, IGNORED_ARG));
+    STRICT_EXPECTED_CALL(malloc_flex(IGNORED_ARG, IGNORED_ARG, sizeof(char)));
+    STRICT_EXPECTED_CALL(mocked_vsnprintf(IGNORED_ARG, IGNORED_ARG, IGNORED_ARG, IGNORED_ARG)).SetReturn(-1);
+    STRICT_EXPECTED_CALL(gballoc_hl_free(IGNORED_ARG));
+
+    // act
+    THANDLE(RC_STRING) rc_string = rc_string_create_with_format("hell%c", 'o');
+
+    // assert
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+    ASSERT_IS_NULL(rc_string);
+}
+
+/* Tests_SRS_RC_STRING_07_008: [ If any error occurs, `rc_string_create_with_format` shall fail and return `NULL`. ]*/
+TEST_FUNCTION(when_underlying_calls_fail_rc_string_create_with_format_also_fails)
```

Comment:
>when_underlying_calls_fail_rc_string_create_with_format_also_fails [](http://example.com/codeflow?start=14&length=66)

We should have a more adversarial test where the return value is -100 for example.